plugins/mcreator-core/blockly/js/mcreator_extensions.js

```diff
@@ -296,6 +296,8 @@ function simpleRepeatingInputMixin(mutatorContainer, mutatorInput, inputName, in
             var connection = containerBlock.getInput('STACK').connection;
             for (let i = 0; i < this.inputCount_; i++) {
                 const inputBlock = workspace.newBlock(mutatorInput);
+                if (fieldNames.length > 0)

```

Comment:
Could you briefly explain what this does?

Reply:
I guess that one could deserve its own PR since this initializes `fieldValues_` property of mutation blocks that append repeating fields when connected to the stack.

Reply:
This is what I was wondering. If this is something needed for all cases, it should be universally fixed

Reply:
Does this PR even need this change given it does not use simpleRepeatingInputMixin?

Reply:
It was needed before I was asked to rewrite the new mutator to not use the helper function for this specific case. Forgot I added this, sorry.

Reply:
@Defeatomizer this was not really resolved so why RFR?

Reply:
Oops 🤭 Don't draft back, will do in a minute or 2.

Reply:
✔️ Fixed